src/functional/ffront/fbuiltins.py

```diff
@@ -75,6 +75,17 @@ def __gt_type__(self):
     )
 )
 
+astype = BuiltInFunction(
+    ct.FunctionType(
+        args=[
+            ct.DeferredSymbolType(constraint=ct.FieldType),
+            ct.DeferredSymbolType(constraint=ct.FunctionType),
```

Comment:
```suggestion
            ct.DeferredSymbolType(constraint=None),
```
as far as I can see that was Till's proposal.

Reply:
Should `astype` also work on scalars? @egparedes

Reply:
We discussed this in a call with @egparedes and @tehrengruber and the conclusion was that it would not